src/components/Results.js

```diff
@@ -12,39 +12,93 @@ const diffDays = (checkInDate, checkOutDate) => {
 //   );
 // };
 
-const Results = props => {
-  return (
-    <table>
-      <thead>
-        <tr>
-          <th>Title</th>
-          <th>First name</th>
-          <th> Surname</th>
-          <th>Email</th>
-          <th>CheckInDate</th>
-          <th>CheckOutDate</th>
-          <th>roomId</th>
-          <th>NumOfDays</th>
-        </tr>
-      </thead>
-      {props.reservations.map(reservation => (
-        <tbody>
-          <tr>
-            <td>{reservation.title}</td>
-            <td>{reservation.firstName}</td>
-            <td>{reservation.surname}</td>
-            <td>{reservation.email}</td>
-            <td>{reservation.checkInDate}</td>
-            <td>{reservation.checkOutDate}</td>
-            <td>{reservation.roomId}</td>
-            <td>
-              {diffDays(reservation.checkInDate, reservation.checkOutDate)}
-            </td>
-          </tr>
-        </tbody>
-      ))}
-    </table>
-  );
-};
+export default class Results extends Component {
+  constructor(props) {
+    console.log("receiving props");
+    console.log(props);
+    super(props);
+    this.state = { reservations: props.reservations };
```

Comment:
This code is correctly using state for the reservations, however I think state is actually unnecessary here. Remember in class how we talked about how we generally prefer to use props, until we **know** that the component needs to update this data when the interacts with it.

Here we are just using the reservations data directly - we aren't calling `this.setState`, so I think props is the better way of doing it.